param/parameterized.py

```diff
@@ -1301,7 +1301,10 @@ class hierarchy (see ParameterizedMetaclass).
         self.precedence = precedence
         self.default = default
         self.doc = doc
-        self.constant = constant is True or readonly is True # readonly => constant
+        if constant is True or readonly is True:  # readonly => constant
+            self.constant = True
+        else:
+            self.constant = constant
```

Comment:
Ooh; good catch! I wonder if there were any other cases where it wouldn't inherit properly due to computation like this.

Reply:
> I wonder if there were any other cases where it wouldn't inherit properly due to computation like this.

Good question 🙃 !